### ISSUE TITLE
Feature request: Documentation for how to check if key exists, or implement the feature

### ISSUE THREAD

**Issue by ryanwalls** [zabawaba99/firego#58]
I can't seem to figure out a clean way to check if a key already exists.  Calling `.Value()` on a reference returns a json error.  I assume I'm just missing something, so perhaps we can add that to the README?

**Comment by zabawaba99** [comment]
I think it would be a good idea to add some functionality to check for existence. Getting a weird json error from `.Value()` seems unexpected.

---

Looking at the [android sdk](https://firebase.google.com/docs/reference/android/com/google/firebase/database/DataSnapshot.html#exists%28%29)  (as well as the javascript one) it looks like there is an `exists()` method on a `DataSnapshot`. The first thought that comes to mind is creating `.Exists()` on the `Firebase` struct. Any thoughts?

**Comment by pmbenjamin** [comment]
The problem seems to be stemming from [line 156](https://github.com/zabawaba99/firego/blob/master/firebase.go#L156) in `firebase.go`.

The function signature indicates:
```
func (fb *Firebase) Value(v interface{}) error
```
But that function returns:
```
return json.Unmarshal(bytes, v)
```

**Comment by toffentoffen** [comment]
I am working on the solutions. Implementing Exists method, and documenting and example.

**Comment by zabawaba99** [comment]
@morfeo8marc Any word on the progress of implementing the `Exists` method?

**Comment by toffentoffen** [comment]
@zabawaba99 It's implemented, but I am struggling with the test case. The mocked server is not marshalling and unmarshalling correctly an empty response. 

> empty response: {}

Empty response is what firebase is returning, instead of not found.

**Comment by zabawaba99** [comment]
If you'd like, you can make a pull request and I can help with writing the test case.

**Comment by tqwewe** [comment]
Came here to request this feature too. Javascript has the .hasChild method. Would be good to see something similar in this Golang package.

**Comment by zabawaba99** [comment]
Closed by #77
